### ISSUE TITLE
Implementing a refactoring of legacy custom-scripts code

### ISSUE THREAD

**Issue by tatmaca** [uchicago-cs/chiventure#1557]
The custom-scripts team decided to refactor the legacy source code, as it was previously all stored in one large, difficult to read source file. After reviewing the code, we decided that we could refactor the source code into four modules: the arguments (arg) module, the object_type (obj) module, the lua module, and the implement into game (get) module.

We decided that the "arg" and "obj" modules would be necessary as it stored the creation and initialization of the two main structs used in the custom-scripts source code. We decided that the "lua" module was necessary as it included all of the unique functions that allowed our code to interact with Lua. We decided that the "get" module was necessary as it stores all the code that members of other developments teams would be interested in looking at in order to get their object_types to interact with their unique custom scripts.

**Comment by tatmaca** [comment]
This issue is now closed. We have implemented the refactoring of the legacy code into four distinct modules (arg, get, obj, and lua). Not only is our code refactored, but we have simplified each of the modules such that our functions are now capable of managing every data type! Next steps from this issue include updating test cases and renaming the functions in our old demonstrations, and then putting in the merge request.
